pkg/log/zap/zap.go

```diff
@@ -82,18 +77,51 @@ type Opts func(*Options)
 // See Options.Development
 func UseDevMode(enabled bool) Opts {
 	return func(o *Options) {
-		o.Development = true
+		o.Development = enabled
```

Comment:
@DirectXMan12 Did you mean to fix this?
https://github.com/kubernetes-sigs/controller-runtime/pull/646#issuecomment-543872875

Reply:
@hasbro17 I noticed this as well, and fixed it and added tests in #653.